### ISSUE TITLE
Nan when trying to replicate AVA results.

### ISSUE THREAD

**Issue by campos-97** [facebookresearch/SlowFast#353]
Hi,

I have been trying to get AVA example running, however, the inference results obtained show weird behavior. Some of the scores show NAN while others show a number as expected. I am only trying to test your pre-trained **SLOWFAST_32x2_R101_50_50_v2.1.pk** model, using only one video of the AVA data set, while maintaining the original action list file.

Please let me know if the error I am experiencing is something expected or if am I doing something wrong.

**To Reproduce**

```
python3 SlowFast/tools/run_net.py \
--cfg SlowFast/configs/AVA/c2/SLOWFAST_32x2_R101_50_50_v2.1.yaml \
NUM_GPUS 1 \
TEST.BATCH_SIZE 1 \
DATA.NUM_FRAMES 16 \
DATA.PATH_TO_DATA_DIR /content/dataset/ \
AVA.FRAME_LIST_DIR /content/dataset/frame_lists/ \
AVA.FRAME_DIR /content/dataset/frames/ \
AVA.ANNOTATION_DIR /content/dataset/annotations/ \
AVA.TEST_PREDICT_BOX_LISTS ['"ava_val_predicted_boxes.csv"'] \
AVA.EXCLUSION_FILE ava_val_excluded_timestamps_v2.1.csv \
AVA.GROUNDTRUTH_FILE ava_val_v2.1.csv \
TEST.CHECKPOINT_FILE_PATH checkpoints/SLOWFAST_32x2_R101_50_50_v2.1.pkl \
TEST.CHECKPOINT_TYPE caffe2
```

**Expected behavior**

```
{ 'PascalBoxes_PerformanceByCategory/AP@0.5IOU/answer phone': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/bend/bow (at the waist)': 0.12757538491869483,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/carry/hold (an object)': 0.6381458902987285,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/climb (e.g., a mountain)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/close (e.g., a door, a box)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/crouch/kneel': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/cut': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/dance': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/dress/put on clothing': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/drink': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/drive (e.g., a car, a truck)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/eat': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/enter': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/fall down': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/fight/hit (a person)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/get up': 0.06666666666666667,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/give/serve (an object) to (a person)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/grab (a person)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/hand clap': 0.0,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/hand shake': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/hand wave': 0.0011160714285714285,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/hit (an object)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/hug (a person)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/jump/leap': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/kiss (a person)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/lie/sleep': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/lift (a person)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/lift/pick up': 0.0036036036036036037,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/listen (e.g., to music)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/listen to (a person)': 0.2505843688555108,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/martial art': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/open (e.g., a window, a car door)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/play musical instrument': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/point to (an object)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/pull (an object)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/push (an object)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/push (another person)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/put down': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/read': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/ride (e.g., a bike, a car, a horse)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/run/jog': 0.3347590432426064,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/sail boat': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/shoot': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/sing to (e.g., self, a person, a group)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/sit': 0.729399856555861,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/smoke': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/stand': 0.7797054286957839,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/swim': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/take (an object) from (a person)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/take a photo': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/talk to (e.g., self, a person, a group)': 0.8068429495584665,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/text on/look at a cellphone': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/throw': 0.0019455252918287938,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/touch (an object)': 0.19910261652055677,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/turn (e.g., a screwdriver)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/walk': 0.7393860040689,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/watch (a person)': 0.3588362134312586,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/watch (e.g., TV)': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/work on a computer': nan,
  'PascalBoxes_PerformanceByCategory/AP@0.5IOU/write': nan,
  'PascalBoxes_Precision/mAP@0.5IOU': 0.33584464154246924}
[12/23 16:07:38][INFO] ava_eval_helper.py: 170: AVA eval done in 6.070424 seconds.
[12/23 16:07:38][INFO] logging.py:  96: json_stats: {"map": 0.33584, "mode": "test"}
```

**Environment**
To perform the tests I am using Google Colab.

**Comment by andres-ramirez27** [comment]
Hi campos-97.  From what I understand about the behavior of the project, those `PascalBoxes_PerformanceByCategory` are like an average (or other operation) of the loss across each one of the frames. You need a ground-truth value and the value predicted by the net to calculate the loss, so if one of them is missing, the result would be NaN, as those you are seeing.

You mentioned that you're only testing with one video, so my guess is that the annotations for that only video doesn't have the ground-truth value for the 80 output classes of the NN as the video doesn't do the 80 actions. As some classes doesn't have that value, loss can't be calculated and NaNs are generated.

If you open the `detections_latest.csv` files that should've been generated at the end of the execution, you'll see that for each frame, you have 80 predictions (one for each class). These are probably the results you're looking for.

**Comment by campos-97** [comment]
Hello andres-ramirez27, 

Thank you very much for your answer. I see what you say and it makes a lot of sense. Now I understand why there are some values that show as NaN. Thanks again for your help.
